mpi-proxy-split/seq_num.cpp

```diff
@@ -274,14 +277,21 @@ void drain_mpi_collective() {
   share_seq_nums(target);
   pthread_mutex_unlock(&seq_num_lock);
   while (1) {
-    dmtcp::string barrierId = "MANA-PRESUSPEND-" + jalib::XToString(round_num);
-    dmtcp::string csId = "MANA-CRITICAL-SECTION-" + jalib::XToString(round_num);
-    dmtcp::string convergeId = "MANA-CONVERGE-" + jalib::XToString(round_num);
-    dmtcp_kvdb64(DMTCP_KVDB_INCRBY, convergeId.c_str(), 0, check_seq_nums(false));
-    dmtcp_kvdb64(DMTCP_KVDB_OR, csId.c_str(), 0, current_phase == IN_CS);
-    dmtcp_global_barrier(barrierId.c_str());
-    dmtcp_kvdb64_get(convergeId.c_str(), 0, &num_converged);
-    dmtcp_kvdb64_get(csId.c_str(), 0, &in_cs);
+    char key[32] = {0};
+    char barrier_id[32] = {0};
+    dmtcp::string cs_id = "/plugin/MANA/CRITICAL-SECTION";
+    dmtcp::string converge_id = "/plugin/MANA/CONVERGE";
+    sprintf(barrier_id, "MANA-PRESUSPEND-%06d", round_num);
```

Comment:
I'm not familiar with the seq algorithm, should it be `MANA/PRESUSPEND-%06d` like `plugin/MANA/CRITICAL-SECTION`?